ambry-api/src/main/java/com.github.ambry/config/RouterConfig.java

```diff
@@ -429,5 +436,6 @@ public RouterConfig(VerifiableProperties verifiableProperties) {
     }
     routerOperationTrackerTerminateOnNotFoundEnabled =
         verifiableProperties.getBoolean("router.operation.tracker.terminate.on.not.found.enabled", false);
+    routerMaxInMemChunks = verifiableProperties.getIntInRange("router.max.in.mem.chunks", 4, 1, Integer.MAX_VALUE);
```

Comment:
Also Integer.MAX_VALUE might not be a valid end for the range. We cannot buffer so much data in memory. Maybe Integer.MAX_VALUE/CHUNK_SIZE(8MB) is a better end?

Reply:
Good suggestion!